packages/frontend/src/app/components/TrustRegistryTable.tsx

```diff
@@ -0,0 +1,72 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHeader,
+  TableRow,
+} from '@/common/components/Table'
+import { Entity } from '@/common/interfaces'
+import Image from 'next/image'
+import dayjs from 'dayjs'
+import { useRouter } from 'next/navigation'
+import { backendUrl, betterFetch } from '@/api'
+
+export function TrustRegistryTable() {
+  const router = useRouter()
+  const [entities, setEntities] = useState<Entity[]>([])
+
+  useEffect(() => {
+    getEntities()
+      .then((entities) => {
+        setEntities(entities)
+      })
+      .catch(() => {
+        setEntities([])
+      })
+  }, [])
+
+  return (
+    <Table>
+      <TableHeader>
+        <th className="p-4 w-8/12">Company name</th>
+        <th className="p-4 w-2/12 text-right">Updated</th>
+        <th className="p-4 w-2/12 text-right">Registered</th>
+      </TableHeader>
+      <TableBody>
+        {entities.map((item: Entity, rowIndex: number) => {
+          return (
+            <TableRow key={rowIndex} onClick={() => router.push(`/${item.id}`)}>
```

Comment:
Can't we use a link for this? Therefore we shouldn't need the client component.

Reply:
Good point 👍

Reply:
I had a look to that and unfortunately I don't see a possibility to use link here because `TableRow` is `<tr>` and it has `onClick` but wrapping all `<td>` inside of it into `<Link>` doesn't seems correct to me.

Reply:
And wrapping `tr` in `Link` is also bad?